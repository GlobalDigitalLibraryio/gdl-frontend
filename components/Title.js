// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import styled from 'styled-components';
import tag from 'tag-hoc';
import { fontSize, color } from 'styled-system';

const Base = tag(['textAlign', 'upperCase'])('h1');

// TODO: Figure out how to type styled components
/* type Props = {
  textAlign: 'left' | 'center' | 'right',
  fontSize: Array<number> | number | Array<string> | string,
  color?: string,
  is?: string,
}; */

const Title = styled(Base)`
  ${fontSize} ${color} display: block;
  font-weight: 600;
  text-align: ${props => props.textAlign};
  text-transform: ${props => (props.upperCase ? 'uppercase' : null)};
  text-decoration: none;
`;

Title.defaultProps = {
  fontSize: [28, 38],
  textAlign: 'left',
};

export default Title;
