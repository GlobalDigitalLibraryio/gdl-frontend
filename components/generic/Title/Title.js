import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.h3`
  word-break: break-word;
  color: #363636;
  font-size: 2rem;
  font-weight: 300;
  line-height: 1.125;
  text-align: ${props => (props.textCentered ? 'center' : 'left')};

  &:not(:last-child) {
    margin-bottom: 1.5em;
  }

  & span,
  & em {
    font-weight: 300;
  }

  & strong {
    color: inherit;
    font-weight: 500;
  }
`;

Title.propTypes = {
  textCentered: PropTypes.bool
};

Title.defaultProps = {
  textCentered: false
};

const H1 = Title.withComponent('h1').extend`
  font-size: 3rem;
`;

const H2 = Title.withComponent('h2').extend`
  font-size: 2.5rem;
`;

// Default
const H3 = Title;

const H4 = Title.withComponent('h4').extend`
  font-size: 1.5rem;
`;

const H5 = Title.withComponent('h5').extend`
  font-size: 1.25rem;
`;

const H6 = Title.withComponent('h6').extend`
  font-size: 1rem;
`;

// When the semantics of h tags aren't correct, but you would still like the styling
const P1 = H1.withComponent('p');
const P2 = H2.withComponent('p');
const P3 = H3.withComponent('p');
const P4 = H4.withComponent('p');
const P5 = H5.withComponent('p');
const P6 = H6.withComponent('p');

export { H1, H2, H3 as default, H4, H5, H6, P1, P2, P3, P4, P5, P6 };
