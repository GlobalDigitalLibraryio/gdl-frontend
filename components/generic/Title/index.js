/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import H3, { H1, H2, H4, H5, H6, P1, P2, P3, P4, P5, P6 } from './Title';

export { H1, H2, H3 as default, H4, H5, H6, P1, P2, P3, P4, P5, P6 };
