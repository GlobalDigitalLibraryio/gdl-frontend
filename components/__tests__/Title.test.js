// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Title from '../Title';

test('Renders a Title', () => {
  const tree = mount(<Title>Normal title</Title>);

  expect(toJson(tree)).toMatchSnapshot();
});

test('Supports uppercasing', () => {
  const tree = mount(<Title upperCase>Uppercased title</Title>);

  // Get the h1 node and check it if has the proper CSS rule
  expect(toJson(tree.find('h1'))).toHaveStyleRule(
    'text-transform',
    'uppercase',
  );
});
