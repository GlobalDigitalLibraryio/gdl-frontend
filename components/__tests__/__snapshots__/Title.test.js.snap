// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a Title 1`] = `
.c0 {
  font-size: 28px;
  display: block;
  margin-top: 0;
  font-weight: 600;
}

@media screen and (min-width:40em) {
  .c0 {
    font-size: 38px;
  }
}

<Styled(Base)
  fontSize={
    Array [
      28,
      38,
    ]
  }
>
  <Base
    className="c0"
    fontSize={
      Array [
        28,
        38,
      ]
    }
  >
    <h1
      className="c0"
      fontSize={
        Array [
          28,
          38,
        ]
      }
      is={null}
    >
      Normal title
    </h1>
  </Base>
</Styled(Base)>
`;
